src/Symfony/Component/Messenger/Adapter/Factory/AdapterFactoryInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Adapter\Factory;
+
+/**
+ * Creates a Messenger adapter.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+interface AdapterFactoryInterface
```

Comment:
The thing is: for the developer experience, I believe that we need to have a very common way of creating the adapters: via a simple DSN, and that's the point of this (thin) adapter factory. That way, it's super easy for other developers to create new adapters.

If we move the `supports` down to the `AdapterInterface`, then we need to move the `$dsn` argument to both the `receiver` and `sender` methods and sort of force a double-parsing of it on the developer's perspective. I don't mean that it's impossible but less elegant I'd say.

Reply:
The parsing could (should ?) be done in the FrameworkExtension ?

Reply:
If it's only for "native" adapters, yes. But this can't be done for adapters coming from 3rd party libraries, and that's the entire point: we want 3rd party component to plug-in and have people using a simple DSN such as `enqueue+sqs://sqs-address` for example. While this wouldn't be in the core obviously.

Reply:
To me, it is still another layer of indirection. But I'm not strongly against this. Let's see what other think.

Reply:
You are right. It's "an extension point" we don't need now as just the adapter factory is enough. Removed in https://github.com/symfony/symfony/pull/26632/commits/d6c046aed17efca71cc112fa4dd88783cfe74119.